src/imitation/scripts/train_preference_comparisons.py

```diff
@@ -24,6 +24,38 @@
 )
 
 
+def save_model(
+    agent_trainer: preference_comparisons.AgentTrainer,
+    vec_normalize: Optional[vec_env.VecNormalize],
+    save_path: str,
+):
+    """Save the model as model.pkl."""
+    serialize.save_stable_model(
+        output_dir=os.path.join(save_path, "policy"),
+        model=agent_trainer.algorithm,
+        vec_normalize=vec_normalize,
+    )
+
+
+def save_checkpoint(
+    trainer: preference_comparisons.PreferenceComparisons,
+    vec_normalize: Optional[vec_env.VecNormalize],
+    save_path: str,
+    allow_save_policy: Optional[bool],
+):
+    """Save reward model and optionally policy."""
+    os.makedirs(save_path, exist_ok=True)
+    th.save(trainer.reward_trainer.model, os.path.join(save_path, "reward_net.pt"))
+    if allow_save_policy:
+        # Note: We should only save the model as model.pkl if `trajectory_generator`
+        # contains one. Specifically we check if the `trajectory_generator` contains an
+        # `algorithm` attribute.
+        assert hasattr(trainer.trajectory_generator, "algorithm")
+        save_model(trainer.trajectory_generator, vec_normalize, save_path)
+    else:
+        print("trainer.trajectory_generator doesn't contain a policy to save.")
```

Comment:
Avoid print statements; use `custom_logger.warn` instead (it'll then get logged to files not just stdout)

Reply:
Good point! I would log the warning messages to trainer.logger.